src/HIE/Bios/Cradle.hs

```diff
@@ -457,13 +456,14 @@ cabalAction work_dir mc l fp = do
 removeInteractive :: [String] -> [String]
 removeInteractive = filter (/= "--interactive")
 
--- Strip out any ["+RTS", ..., "-RTS"] sequences in the command string list.
+-- | Strip out any ["+RTS", ..., "-RTS"] sequences in the command string list.
 removeRTS :: [String] -> [String]
-removeRTS ("+RTS" : xs)  =
-  case dropWhile (/= "-RTS") xs of
-    [] -> []
-    (_ : ys) -> removeRTS ys
-removeRTS (y:ys)         = y : removeRTS ys
+removeRTS (y:ys)
+  -- rts flags are spanning multiple entries
+  | y == "+RTS" = removeRTS (tail $ dropWhile (/= "-RTS") ys)
+  -- rts flags are in a single string
+  | "+RTS" `isPrefixOf` y && "-RTS" `isSuffixOf` y = removeRTS ys
```

Comment:
I'm not sure that we should be checking the suffix. I don't know what to do in the case that `ys` is non-empty and `"-RTS"` isn't a suffix, but if `ys` is empty, we shouldn't need the `"-RTS"` suffix.

Reply:
Thank you for your review! 

I think it is kind of debatable what to do.
Theoretically, it is possible to have a `+RTS` without a `-RTS` if it is the last argument. 
But if there are flags such as `["+RTS -H32m", "-RTS"]` I think we should just let it crash unless we see that we need to fix it.

Reply:
Ah, sorry I didn't see your comment before my suggestion. Makes sense!